### PR TITLE
New Acunetix 360 parser

### PR DESCRIPTION
**Pull request by kiliczsh**
Acunetix 360 is web application version for Acunetix which is hosted at https://online.acunetix360.com/ so that it needs another Scan Type.

### CODE REVIEW

**Comment by kiliczsh** [comment]
Failing step is timeout issue @damiencarol @kiblik 
```
Run helm install \
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/runner/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /home/runner/.kube/config
Error: Internal error occurred: failed calling webhook "validate.nginx.ingress.kubernetes.io": Post https://ingress-nginx-controller-admission.kube-system.svc:443/extensions/v1beta1/ingresses?timeout=30s: dial tcp 10.97.175.172:443: i/o timeout
Error: Process completed with exit code 1.
```

File: dojo/tools/acunetix360/parser.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            sev = item["Severity"]
+            if sev not in ['Info', 'Low', 'Medium', 'High', 'Critical']:
+                sev = 'Info'
+            mitigation = cleantags(item["RemedialProcedure"])
+            references = cleantags(item["RemedyReferences"])
+            url = item["Url"]
+            impact = cleantags(item["Impact"])
+            dupe_key = title
+            request = item["HttpRequest"]["Content"]
+            response = item["HttpResponse"]["Content"]
```
Comment (valentijnscholten) [inline dojo/tools/acunetix360/parser.py:59]: are these values always there? might be better to check for presence of the attributes everywhere before accessing them

**Review by valentijnscholten — CHANGES_REQUESTED** [review]

File: dojo/tools/acunetix360/parser.py (hunk 1 above)
Comment (kiliczsh) [reply dojo/tools/acunetix360/parser.py]: Added None check for request and response fields with default values.

File: dojo/tools/acunetix360/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+__author__ = 'kiliczsh'
```
Comment (damiencarol) [inline dojo/tools/acunetix360/__init__.py:1]:
```suggestion
```
We don't rely on meta attribute to track ownership of code but on VCS

File: dojo/fixtures/test_type.json [hunk 3]
```diff
@@ -726,5 +726,12 @@
         },
         "model": "dojo.test_type",
         "pk": 195
+    },
+    {
+        "fields": {
+            "name": "Acunetix360 Scan"
+        },
+        "model": "dojo.test_type",
+        "pk": 196
```
Comment (damiencarol) [inline dojo/fixtures/test_type.json:735]: This is not needed and will be removed soon.

File: dojo/tools/acunetix360/parser.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import json
+import re
+
+from dojo.models import Finding, Endpoint
+
+__author__ = "Muhammed Kılıç"
+__status__ = "Development"
```
Comment (damiencarol) [inline dojo/tools/acunetix360/parser.py:7]:
```suggestion
```
We don't rely on meta attribute to track authorship of code but we use the VCS

File: dojo/tools/acunetix360/parser.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Acunetix360 JSON format."
+
+    def get_findings(self, filename, test):
+        tree = filename.read()
+        try:
+            data = json.loads(str(tree, 'utf-8-sig'))
+        except:
+            data = json.loads(tree)
```
Comment (damiencarol) [inline dojo/tools/acunetix360/parser.py:35]:
```suggestion
        data = json.load(filename)
```

File: dojo/tools/acunetix360/parser.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+        for item in data["Vulnerabilities"]:
+            mitigation = ''
+            impact = ''
+            references = ''
+            findingdetail = ''
+            title = ''
+            request = ''
+            response = ''
+
```
Comment (damiencarol) [inline dojo/tools/acunetix360/parser.py:46]:
```suggestion
```
These lines are not needed as you redefine the attributes later

**Review by damiencarol — CHANGES_REQUESTED** [review]

File: dojo/tools/acunetix360/parser.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+            finding = Finding(title=title,
+                              test=test,
+                              description=findingdetail,
+                              severity=sev.title(),
+                              mitigation=mitigation,
+                              impact=impact,
+                              references=references,
+                              cwe=cwe,
+                              static_finding=True)
+
```
Comment (damiencarol) [inline dojo/tools/acunetix360/parser.py:74]: There is a date in the reports, you can use it to set `date` attribute.

Comment (kiliczsh) [reply dojo/tools/acunetix360/parser.py:60]: I am not sure about formatting so would be grateful if you help us with using `item["Generated"]` value @damiencarol

Comment (damiencarol) [reply dojo/tools/acunetix360/parser.py:58]: I was thinking to use the something like that:
```python
datetime.datetime.strptime(obj["Target"]["Initiated"][0:10], "%d/%m/%Y")
```

Comment (kiliczsh) [reply dojo/tools/acunetix360/parser.py:61]: Used phrase from qualys and result is --> ``` scan_date = datetime.datetime.strptime(item["Generated"], "%Y-%m-%dT%H:%M:%SZ").date() ```

File: dojo/tools/acunetix360/parser.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+from dojo.models import Finding, Endpoint
+
+# Function to remove HTML tags
+TAG_RE = re.compile(r'<[^>]+>')
+
+
+def cleantags(text=''):
+    prepared_text = text if text else ''
+    return TAG_RE.sub('', prepared_text)
```
Comment (damiencarol) [inline dojo/tools/acunetix360/parser.py:13]: We use a specific module to do that `html2text`

File: dojo/tools/acunetix360/parser.py [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+        for item in data["Vulnerabilities"]:
+            title = item["Name"]
+            findingdetail = cleantags(item["Description"])
+            cwe = int(item["Classification"]["Cwe"]) if "Cwe" in item["Classification"] else None
+            sev = item["Severity"]
+            if sev not in ['Info', 'Low', 'Medium', 'High', 'Critical']:
+                sev = 'Info'
+            mitigation = cleantags(item["RemedialProcedure"])
+            references = cleantags(item["RemedyReferences"])
```
Comment (damiencarol) [inline dojo/tools/acunetix360/parser.py:37]:
```suggestion
            mitigation = html2txt.html2text(item.get("RemedialProcedure", ""))
            references = html2txt.html2text(item.get("RemedyReferences", ""))
```

File: dojo/tools/acunetix360/parser.py [hunk 10]
```diff
@@ -0,0 +1,72 @@
+import json
+import datetime
+import html2text
+
+from dojo.models import Finding, Endpoint
+
+
+def cleantags(text=''):
+    prepared_text = text if text else ''
+    return html2text.html2text(prepared_text)
```
Comment (damiencarol) [inline dojo/tools/acunetix360/parser.py:10]:
```suggestion
```
See my comment below

**Review by damiencarol — APPROVED** [review]

**Review by valentijnscholten — APPROVED** [review]

**Pull request merged**